src/Datasource/EntityTrait.php

```diff
@@ -360,6 +360,20 @@ public function unsetProperty($property)
     }
 
     /**
+     * Get the list of all properties.
+     *
+     * The list of all properties is all standard properties
+     * plus virtual properties plus hidden properties.
+     *
+     * @return array A list of all entity properties
+     */
+    public function allProperties()
```

Comment:
@ADmad Wouldn't it be better to stick with the current API, and use `extract()` to actually retrieve data for the properties? Making `allProperties()` work different than the existing `*Properties()` methods feels a little odd.